src/Network/Transport/TCP.hs

```diff
@@ -994,11 +996,11 @@ handleConnectionRequest transport socketClosed (sock, sockAddr) = handle handleE
                 return ourEndPoint
           TransportClosed ->
             throwIO $ userError "Transport closed"
-        void $ go ourEndPoint theirAddress
+        go ourEndPoint theirAddress
 
       where
 
-      go :: LocalEndPoint -> EndPointAddress -> IO ()
+      go :: LocalEndPoint -> EndPointAddress -> IO (Maybe (IO ()))
       go ourEndPoint theirAddress = do
         -- This runs in a thread that will never be killed
```

Comment:
And here's the problem. This has been wrong since #49 was merged. I'll have to rig this up to deal with asynchronous exceptions.